src/test/app/Batch_test.cpp

```diff
@@ -510,207 +533,288 @@ class Batch_test : public beast::unit_test::suite
+            BEAST_EXPECT(env.balance(alice) == preAlice);
+            BEAST_EXPECT(env.balance(alice, USD.issue()) == preAliceUSD);
+            BEAST_EXPECT(env.seq(bob) == preBobSeq);
+            BEAST_EXPECT(env.balance(bob) == preBob);
+            BEAST_EXPECT(env.balance(bob, USD.issue()) == preBobUSD);
+        }
+
+        // Bad Fee Dynamic Fee Calculation
+        {
+            test::jtx::Env env{*this, envconfig()};
+
+            auto const alice = Account("alice");
+            auto const bob = Account("bob");
+            auto const carol = Account("carol");
+            auto const gw = Account("gw");
+            auto const USD = gw["USD"];
+
+            env.fund(XRP(1000), alice, bob, carol, gw);
+            env.close();
+            env.trust(USD(1000), alice, bob);
+            env(pay(gw, alice, USD(100)));
+            env(pay(gw, bob, USD(100)));
+            env.close();
+
+            auto const preAlice = env.balance(alice);
+            auto const preBob = env.balance(bob);
+
+            auto const seq = env.seq(alice);
+            auto const ammCreate =
+                [&alice](STAmount const& amount, STAmount const& amount2) {
+                    Json::Value jv;
+                    jv[jss::Account] = alice.human();
+                    jv[jss::Amount] = amount.getJson(JsonOptions::none);
+                    jv[jss::Amount2] = amount2.getJson(JsonOptions::none);
+                    jv[jss::TradingFee] = 0;
+                    jv[jss::TransactionType] = jss::AMMCreate;
+                    return jv;
+                };
+
+            auto const batchFee = calcBatchFee(env, 0, 2);
+            env(batch::batch(alice, seq, batchFee, tfAllOrNothing),
+                batch::add(ammCreate(XRP(10), USD(10)), seq + 1),
+                batch::add(pay(alice, bob, XRP(10)), seq + 2),
+                ter(telINSUF_FEE_P));
+            env.close();
+
+            BEAST_EXPECT(env.seq(alice) == seq);
+            BEAST_EXPECT(env.balance(alice) == preAlice);
+            BEAST_EXPECT(env.balance(bob) == preBob);
+        }
     }
 
     void
-    testBadFeeSigner(FeatureBitset features)
+    testChangesBetweenViews(FeatureBitset features)
     {
-        testcase("bad fee signer");
+        testcase("changes between views");
 
         using namespace test::jtx;
         using namespace std::literals;
 
         test::jtx::Env env{*this, envconfig()};
 
-        auto const feeDrops = env.current()->fees().base;
         auto const alice = Account("alice");
         auto const bob = Account("bob");
-        auto const gw = Account("gw");
-        auto const USD = gw["USD"];
-
-        env.fund(XRP(1000), alice, bob, gw);
-        env.close();
-        env.trust(USD(1000), alice, bob);
-        env(pay(gw, alice, USD(100)));
-        env(pay(gw, bob, USD(100)));
-        env.close();
-
-        env(noop(bob), ter(tesSUCCESS));
+        auto const carol = Account("carol");
+        env.fund(XRP(220), alice, bob, carol);
```

Comment:
would prefer not hardcoding the values since they are subjected to change, and it's hard for the reviewer to know the exact numbers. Should use following variables for  account funding:
```
            auto const acctReserve = env.current()->fees().accountReserve(0);
            auto const incReserve = env.current()->fees().increment;
            auto const fee = env.current()->fees().base;
```

Reply:
(feel free to leave these test-related comments for later, since they are less urgent)

Reply:
fixed